internal/bminventory/inventory.go

```diff
@@ -2103,60 +2094,65 @@ func (b *bareMetalInventory) updateHostRoles(ctx context.Context, params install
+		var host, err = getHost(hostName.ID.String(), params.ClusterID.String(), db, log)
 		if err != nil {
-			log.WithError(err).Errorf("failed to find host <%s> in cluster <%s>",
-				params.ClusterUpdateParams.HostsNames[i].ID, params.ClusterID)
 			return common.NewApiError(http.StatusNotFound, err)
 		}
-		if err = hostutil.ValidateHostname(params.ClusterUpdateParams.HostsNames[i].Hostname); err != nil {
+		if err = hostutil.ValidateHostname(hostName.Hostname); err != nil {
 			log.WithError(err).Errorf("invalid hostname format: %s", err)
 			return err
 		}
-		err = b.hostApi.UpdateHostname(ctx, &host, params.ClusterUpdateParams.HostsNames[i].Hostname, db)
+		err = b.hostApi.UpdateHostname(ctx, host, hostName.Hostname, db)
 		if err != nil {
 			log.WithError(err).Errorf("failed to set hostname <%s> host <%s> in cluster <%s>",
-				params.ClusterUpdateParams.HostsNames[i].Hostname, params.ClusterUpdateParams.HostsNames[i].ID,
+				hostName.Hostname, hostName.ID,
 				params.ClusterID)
 			return common.NewApiError(http.StatusConflict, err)
 		}
 	}
 	return nil
 }
 
+func getHost(hostId string, clusterId string, db *gorm.DB, log logrus.FieldLogger) (*models.Host, error){
+	var host models.Host
+	err := db.First(&host, "id = ? and cluster_id = ?", hostId, clusterId).Error
+	if err != nil {
+		log.WithError(err).Errorf("failed to find host <%s> in cluster <%s>", hostId, clusterId)
+		return nil, err
+	}
+	return &host, nil
+}
+
 func (b *bareMetalInventory) updateHostsDiskSelection(ctx context.Context, params installer.UpdateClusterParams, db *gorm.DB, log logrus.FieldLogger) error {
 	for i := range params.ClusterUpdateParams.DisksSelectedConfig {
-		var host models.Host
-		err := db.First(&host, "id = ? and cluster_id = ?",
-			params.ClusterUpdateParams.DisksSelectedConfig[i].ID, params.ClusterID).Error
+		disksConfig := params.ClusterUpdateParams.DisksSelectedConfig[i]
+		hostId := string(disksConfig.ID)
+		host, err := getHost(hostId, string(params.ClusterID), db, log)
+
 		if err != nil {
-			log.WithError(err).Errorf("failed to find host <%s> in cluster <%s>",
-				params.ClusterUpdateParams.DisksSelectedConfig[i].ID, params.ClusterID)
 			return common.NewApiError(http.StatusNotFound, err)
 		}
 
-		var installationDiskPath = ""
-		for _, diskConfig := range params.ClusterUpdateParams.DisksSelectedConfig[i].DisksConfig {
-			if models.DiskRoleInstall == diskConfig.Role {
-				if installationDiskPath != "" {
-					return common.NewApiError(http.StatusConflict,
-						fmt.Errorf("duplicate setting of installation path by the user"))
-				}
-				installationDiskPath = hostutil.FindDiskPathByID(*diskConfig.ID, &host)
-			}
+		disksToInstallOn := funk.Find(disksConfig.DisksConfig, func(diskConfigParams *models.DiskConfigParams) bool {
+			return models.DiskRoleInstall == diskConfigParams.Role
+		}).([]*models.DiskConfigParams)
+
+		installationDiskPath := ""
+
+		if len(disksToInstallOn) > 1{
```

Comment:
file doesn't look like it's been through gofmt `1{`

Reply:
done